libgnucash/engine/Account.cpp

```diff
@@ -5665,38 +5667,36 @@ build_non_bayes (const char *key, const GValue *value, gpointer user_data)
     g_free (guid_string);
 }
 
-static std::tuple<std::string, std::string, std::string>
+static std::tuple<std::string, std::string>
 parse_bayes_imap_info (std::string const & imap_bayes_entry)
 {
-    auto header_length = strlen (IMAP_FRAME_BAYES);
-    std::string header {imap_bayes_entry.substr (0, header_length)};
     auto guid_start = imap_bayes_entry.size() - GUID_ENCODING_LENGTH;
-    std::string keyword {imap_bayes_entry.substr (header_length + 1, guid_start - header_length - 2)};
+    std::string keyword {imap_bayes_entry.substr (1, guid_start - 2)};
     std::string account_guid {imap_bayes_entry.substr (guid_start)};
-    return std::tuple <std::string, std::string, std::string> {header, keyword, account_guid};
+    return std::tuple <std::string, std::string> {keyword, account_guid};
 }
 
 static void
-build_bayes (const char *key, KvpValue * value, GncImapInfo & imapInfo)
+build_bayes (const char *suffix, KvpValue * value, GncImapInfo & imapInfo)
 {
-    auto parsed_key = parse_bayes_imap_info (key);
+    auto parsed_key = parse_bayes_imap_info (suffix);
     GncGUID guid;
     try
     {
-        auto temp_guid = gnc::GUID::from_string (std::get <2> (parsed_key));
+        auto temp_guid = gnc::GUID::from_string (std::get <1> (parsed_key));
```

Comment:
No need for `temp_guid`, assign directly to `guid`.

Reply:
Done